core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1074,23 +1075,92 @@ private static RuntimeException notComparable(String op, Object b0,
   }
 
   // &
-  /** Helper function for implementing <code>BIT_AND</code> */
+  /** Helper function for implementing <code>BIT_AND</code> applied to integer values */
   public static long bitAnd(long b0, long b1) {
     return b0 & b1;
   }
 
+  /** Helper function for implementing <code>BIT_AND</code> applied to binary values */
+  public static ByteString bitAnd(ByteString b0, ByteString b1) {
+
+    if (b0.length() == 0) {
+      return b1;
+    }
+    if (b1.length() == 0) {
+      return b0;
+    }
+
+    if (b0.length() != b1.length()) {
+      throw RESOURCE.differentLengthForBitwiseOperands(b0.length(), b1.length()).ex();
+    }
+    byte[] bytes0 = b0.getBytes();
+    byte[] bytes1 = b1.getBytes();
+    byte[] result = new byte[b0.length()];
+
+    IntStream.range(0, bytes0.length).forEach(i -> {
+      result[i] = (byte) (bytes0[i] & bytes1[i]);
+    });
+    return new ByteString(result);
+  }
+
   // |
-  /** Helper function for implementing <code>BIT_OR</code> */
+  /** Helper function for implementing <code>BIT_OR</code> applied to integer values */
   public static long bitOr(long b0, long b1) {
     return b0 | b1;
   }
 
+  /** Helper function for implementing <code>BIT_OR</code> applied to binary values */
+  public static ByteString bitOr(ByteString b0, ByteString b1) {
+
+    if (b0.length() == 0) {
+      return b1;
+    }
+    if (b1.length() == 0) {
+      return b0;
+    }
+
+    if (b0.length() != b1.length()) {
+      throw RESOURCE.differentLengthForBitwiseOperands(b0.length(), b1.length()).ex();
+    }
+    byte[] bytes0 = b0.getBytes();
+    byte[] bytes1 = b1.getBytes();
+    byte[] result = new byte[b0.length()];
+
+    IntStream.range(0, bytes0.length).forEach(i -> {
```

Comment:
Should we move it to utility function. Seems like repetitive.

Reply:
Sorry for reply later. The variable definition of the function is the same, but the main processing mode is different between BIT_AND, BIT_OR, BIT_XOR which are &, | and ^. So We don't need to move it to utility function.

Reply:
I think we should do.

Reply:
Thanks @hsyuan @ritesh-kapoor. Yes, moved it to utility function looks more better. Thank you help me to have a look in your free time.